src/tools/dev/scripts/bv_support/bv_fms.sh

```diff
@@ -89,18 +89,18 @@ function bv_fms_dry_run
 function build_fms
 {
     #
-    # Prepare build dir
+    # Uncompress the source file
     #
-    prepare_build_dir $FMS_BUILD_DIR $FMS_FILE
+    uncompress_src_file $FMS_SRC_DIR $FMS_FILE
     untarred_fms=$?
     if [[ $untarred_fms == -1 ]] ; then
-        warn "Unable to prepare FMS build directory. Giving Up!"
+        warn "Unable to uncompress FMS source file. Giving Up!"
```

Comment:
Ok, here it does look like you adjusted `build directory` to `source file`. I am not looking too closely but might wanna make sure you've corrected this everywhere.